### PR TITLE
Read receipt update screen

### PR DESCRIPTION
**Pull request by michaelkirk**
screen shot 2017-09-29 at 12 23 14 pm

screen shot 2017-09-29 at 12 24 01 pm

PTAL @charlesmchen

### CODE REVIEW

File: Signal/src/ViewControllers/ExperienceUpgradesPageViewController.swift [hunk 1]
```diff
@@ -4,6 +4,160 @@
...
+        bodyLabel.numberOfLines = 0
+        bodyLabel.lineBreakMode = .byWordWrapping
+        bodyLabel.textAlignment = .center
+
+        // Image
+        let imageView = UIImageView(image: image)
+        view.addSubview(imageView)
+        imageView.contentMode = .scaleAspectFit
+
+        let buttonTitle = NSLocalizedString("UPGRADE_EXPERIENCE_INTRODUCING_READ_RECEIPTS_PRIVACY_SETTINGS", comment: "button label shown one time, after upgrade")
```
Comment (charlesmchen) [inline Signal/src/ViewControllers/ExperienceUpgradesPageViewController.swift]: Reminder: sync l10n.

**Comment by charlesmchen** [comment]
screenshots?

File: Signal/src/ViewControllers/ExperienceUpgradesPageViewController.swift [hunk 2]
```diff
@@ -4,6 +4,160 @@
...
+
+        // be more forgiving with the tappable area
+        let extraPadding: CGFloat = 20
+        let labelSize = titleLabel.intrinsicContentSize
+        return CGSize(width: labelSize.width + extraPadding, height: labelSize.height + extraPadding)
+    }
+
+    override func layoutSubviews() {
+        super.layoutSubviews()
+        titleLabel?.preferredMaxLayoutWidth = titleLabel?.frame.size.width ?? 0
```
Comment (charlesmchen) [inline Signal/src/ViewControllers/ExperienceUpgradesPageViewController.swift]: Why is this necessary?

File: Signal/src/ViewControllers/ExperienceUpgradesPageViewController.swift [hunk 3]
```diff
@@ -4,6 +4,160 @@
...
+    // MARK: - Overrides
+
+    override var intrinsicContentSize: CGSize {
+        guard let titleLabel = titleLabel else {
+            return CGSize.zero
+        }
+
+        // be more forgiving with the tappable area
+        let extraPadding: CGFloat = 20
+        let labelSize = titleLabel.intrinsicContentSize
```
Comment (charlesmchen) [inline Signal/src/ViewControllers/ExperienceUpgradesPageViewController.swift]: Given that the title label's contents can line wrap, don't we need to measure its size in light of a max width constraint?

**Comment by michaelkirk** [comment]
> screenshots?

updated description with screenshots.

**Comment by michaelkirk** [comment]
In 87e1409
